app/public/js/location_search_map.js

```diff
@@ -1,28 +1,69 @@
+
 function initMap() {
+	//current user LatLng
 	var userPoint = {lat: user_latitude, lng: user_longitude };
-	var map = new google.maps.Map(document.getElementById('map'), {
+	var bounds = new google.maps.LatLngBounds();
+
+	//Map div
+	var mapDiv = document.getElementById('map');
+
+    //Map options
+	var options = {
 	  zoom: 12,
 	  center: userPoint,
 	  draggable: false
-	});
+	};
+	//creating the map
+	var map = new google.maps.Map(mapDiv, options);
+
 	locations.forEach(function(location) {
+
+		var myLatLng = new google.maps.LatLng(location.latitude,location.longitude);
+
 		var locationMarker = new google.maps.Marker({
 		  position: {lat: location.latitude, lng: location.longitude},
 		  map: map,
 		  title: location.name
 		});
+
 		google.maps.event.addListener(locationMarker, 'click', function() {
 			window.location.href = '/location-report/' + location.id;
-		});		
+		});	
+
+		bounds.extend(myLatLng);
+	
 	});
+
+	map.fitBounds(bounds);
+
 	var centreMarker = new google.maps.Marker({
 	  position: userPoint,
 	  map: map,
 	  icon: 'http://maps.google.com/mapfiles/ms/icons/blue-dot.png'
 	});
+    
+    //convert kilometers to meters
+	var radius =search_radius*1000;
+
+	var radius = new google.maps.Circle({
```

Comment:
This redeclares the variable "radius" which can be a little confusing.  Use a new name like "circle" for the google.maps.Circle object and calling circle.setMap(map).